docs/configuration/listeners/stats.rst

```diff
@@ -17,6 +17,7 @@ Every listener has a statistics tree rooted at *listener.<address>.* with the fo
    ssl.handshake, Counter, Total successful TLS connection handshakes
    ssl.session_reused, Counter, Total successful TLS session resumptions
    ssl.no_certificate, Counter, Total successul TLS connections with no client certificate
+   ssl.fail_no_sni_match, Counter, Total TLS connections that were rejected because of missing SNI match
```

Comment:
you will need to merge master and make a separate data-plane-api PR on this.

Reply:
Done (pending merge in [data-plane-api](https://github.com/envoyproxy/data-plane-api)).